src/pip/_internal/commands/show.py

```diff
@@ -27,25 +31,47 @@ class ShowCommand(Command):
 
     def __init__(self, *args, **kw):
         super(ShowCommand, self).__init__(*args, **kw)
-        self.cmd_opts.add_option(
+
+        cmd_opts = self.cmd_opts
+
+        cmd_opts.add_option(
             '-f', '--files',
             dest='files',
             action='store_true',
             default=False,
             help='Show the full list of installed files for each package.')
+        cmd_opts.add_option(
+            '--format',
+            action='store',
+            dest='show_format',
+            default="header",
+            choices=('header', 'json'),
+            help="Select the output format among: header (default) or json",
```

Comment:
The help formatter is of type `UpdatingDefaultsHelpFormatter`; does it not currently show the available/default options?

Reply:
Hi @weakcamel, The help string appears when the help option is called on show:

<img width="803" alt="screen shot 2019-02-19 at 15 15 10" src="https://user-images.githubusercontent.com/15032115/53025929-d4e4aa00-3459-11e9-8df9-70aee2566f77.png">

Users can also call help on the format option itself and the choices are listed:

<img width="539" alt="screen shot 2019-02-19 at 15 18 09 1" src="https://user-images.githubusercontent.com/15032115/53026092-27be6180-345a-11e9-8540-501e1c12eb07.png">

Reply:
Huh, interesting. The classical [argparse.ArgumentDefaultsHelpFormatter](https://docs.python.org/3.7/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter) seems to be handling it in a more helpful way.

Not an issue then - in this case it's probably better to leave list those options as you already do 👍